static/app/views/performance/cache/tables/spanSamplesTable.tsx

```diff
@@ -0,0 +1,170 @@
+  | 'transaction.duration';
+
+type DataRow = Pick<IndexedResponse, DataRowKeys> & {'transaction.duration': number};
+
+type Column = GridColumnHeader<ColumnKeys>;
+
+const COLUMN_ORDER: Column[] = [
+  {
+    key: SpanIndexedField.ID,
+    name: t('Span ID'),
+    width: 150,
+  },
+  {
+    key: SpanIndexedField.SPAN_DESCRIPTION,
+    name: t('Key'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: 'transaction.duration',
+    name: t('Txn Duration'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: SpanIndexedField.CACHE_ITEM_SIZE,
+    name: t('Value Size'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  {
+    key: SpanIndexedField.CACHE_HIT,
+    name: t('Status'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+];
+
+interface Props {
+  data: DataRow[];
+  isLoading: boolean;
+  error?: Error | null;
+  highlightedSpanId?: string;
+  meta?: EventsMetaType;
+  onSampleMouseOut?: ComponentProps<typeof GridEditable>['onRowMouseOut'];
+  onSampleMouseOver?: ComponentProps<typeof GridEditable>['onRowMouseOver'];
+}
+
+export function SpanSamplesTable({
+  data,
+  isLoading,
+  error,
+  meta,
+  onSampleMouseOver,
+  onSampleMouseOut,
+  highlightedSpanId,
+}: Props) {
+  const location = useLocation();
+  const organization = useOrganization();
+
+  return (
+    <GridEditable
+      aria-label={t('Span Samples')}
+      isLoading={isLoading}
+      error={error}
+      data={data}
+      columnOrder={COLUMN_ORDER}
+      columnSortBy={[]}
+      grid={{
+        renderHeadCell: col =>
+          renderHeadCell({
+            column: col,
+            location,
+          }),
+        renderBodyCell: (column, row) =>
+          renderBodyCell(column, row, meta, location, organization),
+      }}
+      highlightedRowKey={data.findIndex(row => row.span_id === highlightedSpanId)}
+      onRowMouseOver={onSampleMouseOver}
+      onRowMouseOut={onSampleMouseOut}
+      location={location}
+    />
+  );
+}
+
+function renderBodyCell(
+  column: Column,
+  row: DataRow,
+  meta: EventsMetaType | undefined,
+  location: Location,
+  organization: Organization
+) {
+  if (column.key === SpanIndexedField.ID) {
+    return (
+      <SpanIdCell
+        projectSlug={row.project}
+        traceId={row.trace}
+        timestamp={row.timestamp}
+        transactionId={row[SpanIndexedField.TRANSACTION_ID]}
+        spanId={row[SpanIndexedField.ID]}
+      />
+    );
+  }
+
+  if (column.key === SpanIndexedField.SPAN_DESCRIPTION) {
+    const cacheKey = row[column.key].split(' ')[1]; // TODO - test with multiple keys
+    return <SpanDescriptionCell>{cacheKey}</SpanDescriptionCell>;
+  }
+
+  if (column.key === SpanIndexedField.CACHE_HIT) {
+    return <CacheHitMissCell hit={row[column.key]} />;
+  }
+
+  if (column.key === 'transaction.duration') {
+    return <DurationCell milliseconds={row[column.key]} />;
+  }
+
+  if (column.key === SpanIndexedField.CACHE_ITEM_SIZE) {
+    return <ResourceSizeCell bytes={row[column.key]} />;
+  }
```

Comment:
Would be better to provide a `meta=` to the table, and use the meta for rendering! Samples don't have meta of their own, but you can provide it! There's an example in the Requests module code

Reply:
done!